ocrd_tesserocr/deskew.py

```diff
@@ -0,0 +1,216 @@
+                    content=to_xml(pcgts))
+
+    def _process_segment(self, tessapi, segment, image, xywh, where, page_id, file_id):
+        comments = 'cropped'
+        angle = 0.
+        tessapi.SetImage(image)
+        #tessapi.SetPageSegMode(PSM.AUTO_OSD)
+        #
+        # orientation/script
+        #
+        osr = tessapi.DetectOrientationScript()
+        if osr:
+            assert osr['orient_conf'] and not math.isnan(osr['orient_conf']), \
+                "orientation detection failed (Tesseract probably compiled without legacy OEM, or osd model not installed)"
+            if osr['orient_conf'] < 10:
+                LOG.info('ignoring OSD orientation result %d° due to low confidence %.0f in %s',
+                         osr['orient_deg'], osr['orient_conf'], where)
+            else:
+                LOG.info('applying OSD orientation result %d° with high confidence %.0f in %s',
+                         osr['orient_deg'], osr['orient_conf'], where)
+                angle = osr['orient_deg']
+                if angle:
+                    comments += ',rotated-%d' % angle
+            assert osr['script_conf'] and not math.isnan(osr['script_conf']), \
+                "script detection failed (Tesseract probably compiled without legacy OEM, or osd model not installed)"
+            if osr['script_conf'] < 10:
+                LOG.info('ignoring OSD script result "%s" due to low confidence %.0f in %s',
+                         osr['script_name'], osr['script_conf'], where)
+            else:
+                LOG.info('applying OSD script  result "%s" with high confidence %.0f in %s',
+                         osr['script_name'], osr['script_conf'], where)
+                segment.set_primaryScript(osr['script_name'])
+        else:
+            LOG.warning('no OSD result in %s', where)
+        #
+        # orientation/skew
+        #
+        layout = tessapi.AnalyseLayout()
+        if layout:
+            orientation, writing_direction, textline_order, deskew_angle = layout.Orientation()
+            LOG.info('orientation/deskewing for %s: %s / %s / %s / %.3f', where,
+                      membername(Orientation, orientation),
+                      membername(WritingDirection, writing_direction),
+                      membername(TextlineOrder, textline_order),
+                      deskew_angle)
+            # clockwise rotation, as defined in Tesseract OrientationIdToValue:
+            angle2 = {
+                Orientation.PAGE_RIGHT: 270,
+                Orientation.PAGE_DOWN: 180,
+                Orientation.PAGE_LEFT: 90
+            }.get(orientation, 0)
+            if angle2 != angle:
+                LOG.warning('inconsistent angles from layout analysis (%d) and orientation detection (%d) in %s',
+                            angle2, angle, where)
+            deskew_angle *= - 180 / math.pi
+            if int(deskew_angle):
+                comments += ',deskewed'
+            # if angle:
```

Comment:
Can the commented section be removed?

Reply:
This is merely for documentation. I will improve the comments.